--- a/techpack/display/msm/sde/sde_hw_dsc_1_2.c
+++ b/techpack/display/msm/sde/sde_hw_dsc_1_2.c
@@ -80,5 +80,5 @@ static int _dsc_calc_ob_max_addr(struct sde_hw_dsc *hw_dsc, int num_ss)
 static inline int _dsc_subblk_offset(struct sde_hw_dsc *hw_dsc, int s_id,
 #else
-static inline _dsc_subblk_offset(struct sde_hw_dsc *hw_dsc, int s_id,
+static inline int _dsc_subblk_offset(struct sde_hw_dsc *hw_dsc, int s_id,
 #endif
		u32 *idx)
--- a/techpack/display/msm/sde/sde_hw_vdc.c
+++ b/techpack/display/msm/sde/sde_hw_vdc.c
@@ -67,5 +67,5 @@
 static inline int _vdc_subblk_offset(struct sde_hw_vdc *hw_vdc, int s_id,
 #else
-static inline _vdc_subblk_offset(struct sde_hw_vdc *hw_vdc, int s_id,
+static inline int _vdc_subblk_offset(struct sde_hw_vdc *hw_vdc, int s_id,
 #endif
 		u32 *idx)
