--- a/techpack/audio/dsp/q6afe.c
+++ b/techpack/audio/dsp/q6afe.c
@@ -11061,5 +11061,5 @@ static int afe_get_clk_src(u16 port_id, char *clk_src)
 	}
 
-	if (clkinfo_per_port[idx].clk_src_name == NULL)
+	if (clkinfo_per_port[idx].clk_src_name[0] == '\0')
 		return -EINVAL;
 	strlcpy(clk_src, clkinfo_per_port[idx].clk_src_name,
@@ -11568,10 +11568,11 @@ int afe_set_lpass_clock_v2(u16 port_id,
 		return -EINVAL;
 	}
 
-	if (clk_src_name != NULL) {
+	if (clk_src_name[CLK_SRC_FRACT][0] != '\0' ||
+	    clk_src_name[CLK_SRC_INTEGRAL][0] != '\0') {
 		if (cfg->clk_freq_in_hz % AFE_SAMPLING_RATE_8KHZ) {
-			if (clk_src_name[CLK_SRC_FRACT] != NULL)
+			if (clk_src_name[CLK_SRC_FRACT][0] != '\0')
 				ret = afe_set_source_clk(port_id,
 						clk_src_name[CLK_SRC_FRACT]);
-		} else if (clk_src_name[CLK_SRC_INTEGRAL] != NULL) {
+		} else if (clk_src_name[CLK_SRC_INTEGRAL][0] != '\0') {
 			ret = afe_set_source_clk(port_id,
 					clk_src_name[CLK_SRC_INTEGRAL]);
--- a/techpack/audio/asoc/sec_audio_adaptation.c
+++ b/techpack/audio/asoc/sec_audio_adaptation.c
@@ -689,9 +689,4 @@ static int sec_audio_volume_monitor_data_get(struct snd_kcontrol *kcontrol,
 	}
 
-	if (volume_monitor_value == NULL) {
-		pr_err("%s: volume_monitor_value is NULL\n", __func__);
-		return -EINVAL;
-	}
-
 	for (i = 0; i < VOLUME_MONITOR_GET_PAYLOAD_SIZE; i++)
 		ucontrol->value.integer.value[i] = volume_monitor_value[i];
--- a/techpack/audio/asoc/msm-pcm-routing-v2.c
+++ b/techpack/audio/asoc/msm-pcm-routing-v2.c
@@ -1931,5 +1931,4 @@ static int msm_routing_get_adm_topology(int fedai_id, int session_type,
 			topology = NULL_COPP_TOPOLOGY;
 	}
-done:
 	pr_debug("%s: Using topology %d\n", __func__, topology);
 	return topology;
